src/software/thunderscope/robot_diagnostics/error_log_widgets.py

```diff
@@ -0,0 +1,273 @@
+from pyqtgraph.Qt.QtWidgets import *
+from pyqtgraph.Qt.QtCore import Qt
+from pyqtgraph.Qt import QtGui
+from proto.import_all_protos import *
+from datetime import datetime
+import textwrap
+import software.thunderscope.robot_diagnostics.robot_error_log_icons.error_log_constants as error_constants
+from proto.robot_log_msg_pb2 import RobotLog, LogLevel
+from software.thunderscope.robot_diagnostics.robot_status import RobotStatusView
+from software.thunderscope.constants import LOG_LEVEL_STR_MAP
+
+
+class RobotLogMessageWidget(QFrame):
+    """
+    A single error message from a robot.
+
+    Displays an icon corresponding to the type of error, the robot id, the message
+    and the timestamp of the error
+    """
+
+    def __init__(
+        self, robot_id: int, message: str, icon: QtGui.QPixmap, timestamp=None
+    ):
+        super(RobotLogMessageWidget, self).__init__()
+
+        self.layout = QHBoxLayout()
+        self.icon = QLabel()
+
+        # setting up the icon
+        self.icon_size = self.width() / 10
+        self.icon.setPixmap(
+            icon.scaled(
+                self.icon_size, self.icon_size, Qt.AspectRatioMode.KeepAspectRatio
+            )
+        )
+
+        # info layout contains the robot id and message
+        self.info_layout = QVBoxLayout()
+        self.info_layout.setContentsMargins(
+            0, self.icon_size / 4, 0, self.icon_size / 3
+        )
+
+        # robot id displayed more prominently
+        self.robot_id = QLabel(f"Robot {robot_id}")
+        self.robot_id.setStyleSheet(
+            "font-weight: bold;" f"font-size: {int(self.height() / 25)}px;"
+        )
+
+        # message of the error
+        self.message = QLabel(message)
+        self.message.setStyleSheet(f"font-size: {int(self.height() / 35)}px;")
+
+        self.info_layout.addWidget(self.robot_id)
+        self.info_layout.addWidget(self.message)
+
+        # layout for timestamp info
+        self.time_layout = QVBoxLayout()
+        self.time_layout.setContentsMargins(
+            0, self.icon_size / 4, 0, self.icon_size / 3
+        )
+
+        # the timestamp the error took place
+        self.timestamp = timestamp if timestamp else datetime.now()
+        self.timestamp_label = QLabel(self.timestamp.strftime("%H:%M:%S %f"))
```

Comment:
nit: to be it seemed slightly confusing by the 6 digit number represented. maybe using a period could help?!
```suggestion
        self.timestamp_label = QLabel(self.timestamp.strftime("%H:%M:%S.%f"))
```